url85e313f409_edugasser_Ascensor_tgz-pJ8-ascensor_MainJ8/src/ascensor/Main.java

```diff
@@ -0,0 +1,778 @@
+        } 
+     }
+     public void proxima_parada_interior()
+     {
+         /* miramos los destinos de todos los pasajeros y nos quedamos con el siguiente */
+         int i = 0;
+         int piso_aux;
+         if (ascensor.getPasajeros().size() > 0 ){
+            piso_aux = ascensor.getPasajeros().get(0).getPisoDestino();
+            while (i < ascensor.getPasajeros().size())
+            {
+               if (ascensor.getPasajeros().get(i).getPisoDestino() < piso_aux ){
+                  piso_aux = ascensor.getPasajeros().get(i).getPisoDestino();
+               }
+            }
+             /* actualizamos la proxima parada */
+            ascensor.setProximaParada(piso_aux);
+         }  
+     }
+     /* si hay algun pasajeros que quiera bajarse en este piso */
+     public boolean existe_pasajero_irse()
+     {
+         int i = 0;
+         boolean salir = false;
+     
+       while (i < ascensor.getPasajeros().size() && !salir)
+         {   
+            if (ascensor.getPasajeros().get(i).getPisoDestino() == piso_actual)
+             {
+                 salir = true;
+             }
+             i++;
+         }
+         return salir;
+     }
+     public  ArrayList<Cola>  getColaConAlguien()
+     {
+         if (ascensor.getDireccion().equals("ASC"))
+         {
+            return cola_subida;
+         }else{ 
+            return cola_bajada;
+         }
+     }
+     public  ArrayList<Cola>  getColaActual()
+     {
+         if (ascensor.getDireccion().equals("ASC"))
+         {
+            return cola_subida;
+         }else{
+            return cola_bajada;
+         }
+     }
+     
+     public boolean transitorio()
+     {
+      return (cota_transitorio <= number_delayed[traza]);
+      //return true;
+        
+     }
+     
+     public void aceptar_pasajeros ()
+     {
+         /* cola_subida.get(piso_actual).size() mientras haya gente que subir y sean menos de 10 personas en el ascensor */
+         while (ascensor.getNumPasajeros() < MAX_PASAJEROS && getColaActual().get(piso_actual).size()  > 0 )
+         {
+           /* si recogemos a pasajeros de la planta baja, calculamos el tiempo de espera de ese pasajero && !transitorio */
+            if (piso_actual==0 ){
+                number_delayed[traza]++;
+                
+                if (transitorio()){
+                total_delayed[traza] += clock - getColaActual().get(piso_actual).frente().getTiempoEntrada();    
+                }
+            }
+            internas[getColaActual().get(piso_actual).frente().getPisoDestino()] = 1;
+            ascensor.getPasajeros().add(getColaActual().get(piso_actual).frente());    
+            getColaActual().get(piso_actual).desencolar();  
+            
+         }
+         /* si al final no ha quedado nadie en la cola, actualizamos la botonera */
+         if (getColaActual().get(piso_actual).isEmpty()) {         
+             if (getColaActual().get(piso_actual).getDireccion() == "ASC"){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.